test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,615 @@ describe("BaseAxis", () => {
+            let axis = new Plottable.Axis(mockScale, "left");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x") - TestMethods.numAttr(secondAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("offsets additional rows by the placement of colliding rectangles for right orientation", () => {
+            mockScale.setDomainRangeMapping("B", mockScale.scale("A") + 1);
+            let axis = new Plottable.Axis(mockScale, "right");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(secondAnnotationRect, "x") - TestMethods.numAttr(firstAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+        });
+
+      });
+
+      it("hides annotations if rectangles are outside the margin area", () => {
+        let mockScale = new Mocks.Scale<string, number>();
+        let annotatedTicks = ["A", "B", "C"];
+        mockScale.domain(annotatedTicks);
+
+        mockScale.setDomainRangeMapping("A", 50);
+        mockScale.setDomainRangeMapping("B", 51);
+        mockScale.setDomainRangeMapping("C", 150);
+
+        let axis = new Plottable.Axis(mockScale, "bottom");
+        axis.annotatedTicks(annotatedTicks);
+        axis.annotationsEnabled(true);
+        axis.margin(25);
+
+        let svg = TestMethods.generateSVG(300, 300);
+        axis.renderTo(svg);
+
+        axis.content().selectAll(".annotation-rect").each(function() {
+          let annotationRect = d3.select(this);
+          let bbox = this.getBBox();
+          let allowVisible = bbox.x >= 0 && bbox.x + bbox.width <= axis.width() &&
+                             bbox.y >= axis.height() - axis.margin() && bbox.y + bbox.height <= axis.height();
+          let visibilityAttr = allowVisible ? "visible" : "hidden";
+          assert.strictEqual(annotationRect.attr("visibility"), visibilityAttr, "annotation rect inside margin area");
```

Comment:
This is a bit subtle, but changing the "semantic flow" of the test would better represent what is being tested. Currently, the test checks that the "insideness" and visibility of each annotation is the same. While that logic does check the functionality, the following logic would better represent the test title and what the test is looking for:
- Check if the rectangle is inside the margin.
  - If so, assert that it is visible
  - If not, assert that it is not visible

Since this structure better mirrors the logical flow of the test's name, it's easier for developers to understand what it's doing and be confident that the test is checking the correct thing.